### DISCUSSION TITLE
What could happen if I did NOT await a task in an actor handler?

### DISCUSSION THREAD

**Discussion by tjaart** [akkadotnet/akka.net#6679] [Q&A] [answered]
What could happen if I had a async method and I called it in a handler without awaiting it? Could this cause problems?

```cs
private void ReceiveMyMessage(MyMessage msg) 
{
       _ = SomeAsyncMethod();
}
```

**Comment by Arkatufus** [comment, marked as answer]
It really depends on the async code, if it is thread safe and does not have any callback that references any of the actor state then it should be fine.
What happens is that the thread Task will be launched as a detached Task outside of the actor system scheduler and its completion will not be checked. This is akin to declaring your method as `async void`

**Reply by tjaart** [reply]
Thank you. We have a problem with an unresponsive actor. Since this is not the cause we will keep looking, the library code we are not awaiting is indeed thread safe.
